src/legacy/core_plugins/visualizations/public/vis_types/vis_types_service.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// @ts-ignore
+import { defaultFeedbackMessage } from 'ui/vis/default_feedback_message';
+import { getUpdateStatus, Status } from 'ui/vis/update_status';
+// @ts-ignore
+import { updateOldState } from 'ui/vis/vis_update_state';
+// @ts-ignore
+import { VisProvider } from 'ui/vis/index.js';
+// @ts-ignore
+import { VisFactoryProvider } from 'ui/vis/vis_factory';
+import { VisTypesRegistry, VisTypesRegistryProvider } from 'ui/registry/vis_types';
+
+// Vis, VisParams, VisProvider, VisState
+import * as typesFromVis from 'ui/vis/vis';
+// VisualizationController, VisType
+import * as typesFromVisTypes from 'ui/vis/vis_types/vis_type';
+
+/**
+ * Vis Types Service
+ *
+ * @internal
+ */
+export class VisTypesService {
+  public setup() {
+    return {
+      VisProvider,
+      VisFactoryProvider,
+      VisTypesRegistryProvider,
+      defaultFeedbackMessage, // make default in base vis type, or move?
+      updateOldState, // convert to saved object migration?
+      updateStatus: {
+        getUpdateStatus,
+        Status,
+      },
+    };
+  }
+
+  public stop() {
+    // nothing to do here yet
+  }
+}
+
+/** @public */
+export type VisTypesSetup = ReturnType<VisTypesService['setup']>;
+
+/** @public */
+export type Vis = typesFromVis.Vis;
```

Comment:
Is this the best typescript way to re-export types?
Not sure, but curious :)

On one hand it's nicer to `export type` explicitly
On the other hand, having 
`export {Vis, VisParams ...} from 'ui/vis/vis' `
Is clearer.

Reply:
The main reason I didn't initially do `export { Vis, VisParams } from 'ui/vis'` was because of our use of `*.d.ts` type definition files -- I didn't want to blindly export something that's not a type.

e.g. if `ui/vis` exported a `Vis` class, and also had an `index.d.ts` file exporting a `Vis` type, then `import { Vis } from 'plugins/visualizations'` means you could use `Vis` as either a type or a value. 

But I don't have any concerns about changing this for the sake of readability; once I open a subsequent PR that actually moves `ui/vis`, this will be changed anyway so it isn't relying on typedef files... so it's really just a short-term situation regardless.

Reply:
so we need to export every type as well ? or is this just due to this intermediate step ?